pkg/operator/bootstrap.go

```diff
@@ -212,110 +213,69 @@ func RenderBootstrap(
-			},
-		)
+		if infra.Status.PlatformStatus.OpenStack.LoadBalancer != nil {
+			lbType = infra.Status.PlatformStatus.OpenStack.LoadBalancer.Type
+		}
+		manifests = getPlatformManifests(manifests, strings.ToLower(string(configv1.OpenStackPlatformType)), lbType)
 	}
 
 	if infra.Status.PlatformStatus.Ovirt != nil {
-		manifests = append(manifests,
-			manifest{
-				name:     "manifests/on-prem/coredns.yaml",
-				filename: "ovirt/manifests/coredns.yaml",
-			},
-			manifest{
-				name:     "manifests/on-prem/coredns-corefile.tmpl",
-				filename: "ovirt/static-pod-resources/coredns/Corefile.tmpl",
-			},
-			manifest{
-				name:     "manifests/on-prem/keepalived.yaml",
-				filename: "ovirt/manifests/keepalived.yaml",
-			},
-			manifest{
-				name:     "manifests/on-prem/keepalived.conf.tmpl",
-				filename: "ovirt/static-pod-resources/keepalived/keepalived.conf.tmpl",
-			},
-		)
+		if infra.Status.PlatformStatus.Ovirt.LoadBalancer != nil {
+			lbType = infra.Status.PlatformStatus.Ovirt.LoadBalancer.Type
+		}
+		manifests = getPlatformManifests(manifests, strings.ToLower(string(configv1.OvirtPlatformType)), lbType)
 	}
 
-	if infra.Status.PlatformStatus.VSphere != nil && len(infra.Status.PlatformStatus.VSphere.APIServerInternalIPs) > 0 {
-		manifests = append(manifests,
-			manifest{
-				name:     "manifests/on-prem/coredns.yaml",
-				filename: "vsphere/manifests/coredns.yaml",
-			},
-			manifest{
-				name:     "manifests/on-prem/coredns-corefile.tmpl",
-				filename: "vsphere/static-pod-resources/coredns/Corefile.tmpl",
-			},
-			manifest{
-				name:     "manifests/on-prem/keepalived.yaml",
-				filename: "vsphere/manifests/keepalived.yaml",
-			},
-			manifest{
-				name:     "manifests/on-prem/keepalived.conf.tmpl",
-				filename: "vsphere/static-pod-resources/keepalived/keepalived.conf.tmpl",
-			},
-		)
+	if infra.Status.PlatformStatus.VSphere != nil {
+		emptyAPIInternalIPs := len(infra.Status.PlatformStatus.VSphere.APIServerInternalIPs) == 0
+		lbSetAndTypeOpenShiftManaged := (infra.Status.PlatformStatus.VSphere.LoadBalancer != nil && infra.Status.PlatformStatus.VSphere.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault)
+
+		// vSphere allows to use a user managed load balancer by not setting the VIPs in PlatformStatus.
+		// To maintain backward compatibility, we'll follow this logic:
+		// - If the VIPs are not set in PlatformStatus, we assume that the load balancer is managed by a user managed load balancer (previous behavior).
+		// - If the VIPs are set in PlatformStatus, we will check if the LoadBalancerType is set.
+		//   - If the LoadBalancerType is not set, or set to OpenShiftManaged, we will deploy the OpenShift managed load balancer.
+		//   - If the LoadBalancerType is set to another value than the default, we will not deploy the internal load balancer.
+		deployInternalLB := true
+		if emptyAPIInternalIPs || !lbSetAndTypeOpenShiftManaged {
```

Comment:
I just realized this is wrong, I'm working on it now.
If you're reviewing my PR, please ignore the vsphere block for now. Thanks.